docs/source/results.rst

```diff
@@ -97,17 +97,6 @@ checkpointing or early stopping:
         return TrainResult(some_metric, checkpoint_on=metric_a, early_stop_on=metric_b)
 
 
-In the manual loop, checkpoint and early stop is based only on the loss returned. With the `TrainResult` you
```

Comment:
why is this being removed? this whole block is still true

Reply:
Now I'm not sure, if I understood the block correctly. So maybe you can quickly answer me:

Why can I only use the "loss"?
The test here even says: https://github.com/PyTorchLightning/pytorch-lightning/blob/bd35c869ee0b856fc30229ef36957b058fe97461/tests/callbacks/test_early_stopping.py#L141-L143.

And why can't I change that every batch?
Maybe I somehow completely miss the point of the topic. If so I hapilly undo this change.

Reply:
Ok. I revise my argumentation.
I take everything back and claim the opposite.
I think early stopping only using train return dicts doesn't work at all. 
Not even with 'loss'.
The test case I mentioned above has a bug.

It checks whether current_epoch is smaller than max_epoch (presumably to check if early stopping was triggered).
But this check will always be true, since epochs are counted from 0.
I'll make a new issue, since now we are changing code and I guess it also requires discussion which cases should be supported (any keywords as suggested in the test, only 'loss' as by the docs and you or keep it as it is)

Reply:
Here is a minimal example, to show you that it doesn't work with "loss":

```python3
import pytorch_lightning as pl
import torch
from torch import Tensor
from torch.optim import Optimizer
from torch.utils.data import DataLoader


class MyModule(pl.LightningModule):

    def training_step(self, batch, batch_idx):
        losses = [8, 4, 2, 2, 4, 5, 6, 8, 10, 11]
        return {'loss': torch.tensor(float(losses[self.current_epoch]))}

    def configure_optimizers(self):
        pass

    def backward(self, trainer, loss: Tensor, optimizer: Optimizer, optimizer_idx: int) -> None:
        pass


model = MyModule()
early_stopping = pl.callbacks.EarlyStopping(monitor='loss', patience=0)
trainer = pl.Trainer(early_stop_callback=early_stopping, max_epochs=10)
trainer.fit(model, DataLoader(list(range(100))))
print(trainer.current_epoch)
```

This will be current epoch as 9, but should be lower. See (and please reopen) #3199

Reply:
I will remove this change and we can fix the docs in a new pull request, where we fix issue #3199 
That way, we can get through with this pull request faster.